node/src/Pos/Explorer/Txp/Global.hs

```diff
@@ -52,14 +55,33 @@ extraOps (ExplorerExtra em (HM.toList -> histories) balances) =
     map (uncurry GS.PutAddrBalance) (MM.insertions balances) ++
     map GS.DelAddrBalance (MM.deletions balances)
 
--- CSE-203 FIXME Current time is used as timestamp, are you serious?
 applyBlund
     :: (MonadSlots ctx m, EGlobalApplyToilMode ctx m)
     => TxpBlund
     -> m ()
-applyBlund blund = do
-    curTime <- currentTimeSlotting
-    uncurry (eApplyToil curTime) $ blundToAuxNUndoWHash blund
+applyBlund txpBlund = do
+    -- @TxpBlund@ is a block/blund with a reduced set of information required for
+    -- transaction processing. We use it to determine at which slot did a transaction
+    -- occur. TxpBlund has TxpBlock inside. If it's Left, it's a genesis block which
+    -- doesn't contain transactions. It doesn't have a slot, only epoch, but you can
+    -- use e. g. SlotId epoch minBound. If it's Right, you can use headerSlotL lens.
+    --
+    -- type TxpBlund = (TxpBlock, TxpUndo)
+    -- type TxpBlock = Either (Some IsGenesisHeader) (Some IsMainHeader, TxPayload)
+
+    let txpBlock = txpBlund ^. _1
+    let slotId   = case txpBlock of
+            Left gensisBlock -> SlotId
+                                  { siEpoch = gensisBlock ^. epochIndexL
+                                  , siSlot  = UnsafeLocalSlotIndex 0
```

Comment:
> BTW, we actually don't have a minBound for SlotId.

@ksaric we have `minBound` for `LocalSlotIndex`: https://github.com/input-output-hk/cardano-sl/blob/ee50e1d41cd99082dbf3e030e24745e74eefa66e/core/Pos/Core/Slotting.hs#L154

It's better to use it here instead of `UnsafeLocalSlotIndex`, because `UnsafeLocalSlotIndex` is, well, unsafe.
Of course we know that it's safe here, because its argument is always 0, but anyway, when I see a function prefixed with `Unsafe`, I need to think whether it's safe.

Reply:
Ah, I see, that's what you meant. :man_facepalming: 
Fixed.